### PR TITLE
feat: added "allow_reauth" config option for openstack client

### PR DESCRIPTION
**Pull request by singamSrikar**
# Required for all PRs

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #

This solves the issue of "unauthorised access" because of Openstack API access token expiration.

### CODE REVIEW

**Comment by singamSrikar** [comment]
@powersj Hey, please review this PR.

File: plugins/inputs/openstack/README.md [hunk 1]
```diff
@@ -75,6 +75,13 @@ your requirements. This will help with load and cardinality as well.
   username = "admin"
   password = "password"
 
+  ## AllowReauth should be set to true if you grant permission for Gophercloud to
+  ## cache your credentials in memory, and to allow Gophercloud to attempt to
+  ## re-authenticate automatically if/when your token expires.  If you set it to
+  ## false, it will not cache these settings, but re-authentication will not be
+  ## possible.  This setting defaults to false.
+  # allow_reauth = false
+
```
Comment (srebhan) [inline plugins/inputs/openstack/README.md:84]: I'm not sure if we really need to guide this with an option... Are there cases where the token does not expire or where the current behavior is expected and _wanted_?

**Review by srebhan — COMMENTED** [review]
Thanks for this fix @singamSrikar! I wonder if the current behavior is really _wanted_ or if this is simply a bug that should be fixed!?

File: plugins/inputs/openstack/README.md (hunk 1 above)
Comment (singamSrikar) [reply plugins/inputs/openstack/README.md:84]: Thanks for the quick response, the default expiry of Openstack token is one hour however there is a provision to extend it further but it's not advised to do so for security aspects as this uses admin level access.
What I have observed in our environment is for frequency like 30 min interval(which is ideal to monitor some services in Openstack), the token is expired after two sets of data collection and throws an error of unauthorised access.

Comment (powersj) [reply plugins/inputs/openstack/README.md:84]: Thanks for the PR. This sounds similar to how the k8s token is expiring tokens every hour (see https://github.com/influxdata/telegraf/issues/11267). This does sound more like a bug fix and one that should be the default behavior. @srebhan thoughts?

Comment (srebhan) [reply plugins/inputs/openstack/README.md:84]: I agree. I would not wrap it with an option but "do the right thing" (tm) ;-) without forcing the user to enable the right behavior. TL;DR: Remove the option and do the processing as if `allow_reauth=true`.

**Comment by singamSrikar** [comment]
@srebhan updated the code with suggestions, please review. Thanks.

**Review by powersj — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for the quick response @singamSrikar!

**Pull request merged**